gluten-core/src/main/scala/org/apache/gluten/execution/SortUtils.scala

```diff
@@ -23,11 +23,27 @@ import org.apache.spark.sql.execution.{ProjectExec, SortExec, SparkPlan}
 object SortUtils {
   def dropPartialSort(plan: SparkPlan): SparkPlan = plan match {
     case RewrittenNodeWall(p) => RewrittenNodeWall(dropPartialSort(p))
-    case sort: SortExec if !sort.global => sort.child
+    case PartialSortChild(p) => p
```

Comment:
Why we only drop partial sorts? I know it's previous code but I am still curious about that.

Also I'd suggest renaming:

`PartialSortChild(p)` -> `PartialSortLike(child)`
`ProjectChild(p)` -> `ProjectLike(child)`

Reply:
Sort generated by `EnsureRequirements`.
https://github.com/apache/spark/blob/1bfc9c3cbdc5d017fef68b0e84dd7fd22d2fef0f/sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala#L204